src/resolvers/course.ts

```diff
@@ -38,13 +38,13 @@ export const resolvers = {
       courseFile.author = author || courseFile.author;
       if (units) {
         courseFile.units = [];
-        units.forEach(async (unit: any): Promise<void> => {
+        await units.forEach(async (unit: any): Promise<void> => {
```

Comment:
I think we ought to do this as a map in both cases and wrap in a `Promise.all`. I don't _believe_ this will guarantee that we catch every Promise this way. I'd recommend doing something like:
```
courseFile.units = await Promise.all(units.map(...).filter((t) => t))
``` 
That filter should let you dump to a null when we don't want to do a push and not futz around with async reducers. Top-level its still technically O(n) this way, so not a real time increase for less complex code.